File: leetcode/python/0263_ugly_number.py
class Solution(object):
    def isUgly(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n <= 0: return False
        while (n % 5 == 0 or n % 3 == 0 or n % 2 == 0):
            if n % 5 == 0:
                n //= 5
            
            if n % 3 == 0:
                n //= 3

            if n % 2 == 0:
                n //= 2

        return n == 1

File: leetcode/java/0263_ugly_number.java
class Solution {
    public boolean isUgly(int n) {
        if (n == 1) return true;
        if (n == 0) return false;

        while (n % 2 == 0 || n % 3 == 0 || n % 5 == 0) {
            if (n % 2 == 0) n /= 2;
            if (n % 3 == 0) n /= 3;
            if (n % 5 == 0) n /= 5;
        }

        if (n != 1) return false;

        return true;
    }
}